src/Plugin/media/Source/GoogleForm.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\stanford_media\Plugin\media\Source;
+
+use Drupal\media\MediaInterface;
+use Drupal\media\MediaSourceBase;
+use Drupal\media\MediaSourceFieldConstraintsInterface;
+
+/**
+ * Provides media source plugin for Google Forms iframe.
+ *
+ * @MediaSource(
+ *   id = "google_form",
+ *   label = @Translation("Google Form"),
+ *   description = @Translation("Embed a google form."),
+ *   allowed_field_types = {"string"},
+ *   default_thumbnail_filename = "generic.png"
+ * )
+ */
+class GoogleForm extends MediaSourceBase implements MediaSourceFieldConstraintsInterface {
+
+  /**
+   * Key for "Form ID" metadata attribute.
+   *
+   * @var string
+   */
+  const METADATA_ATTRIBUTE_ID = 'id';
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getMetadataAttributes() {
+    return [
+      self::METADATA_ATTRIBUTE_ID => $this->t('Form ID'),
+    ];
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getMetadata(MediaInterface $media, $attribute_name) {
+    switch ($attribute_name) {
+      case self::METADATA_ATTRIBUTE_ID:
+        return $this->getId($media);
+
+    }
+    return parent::getMetadata($media, $attribute_name);
```

Comment:
Why the switch statement and no default?

Reply:
i've seen the switch as a normal convention on these methods within other media types. for the quick addition of other metadata. guess the bottom line there acts as the `default` case

Reply:
Not your normal style but not unreadable or a performance hit. Meh.